### PR TITLE
Topic5 example use

### LINKED ISSUE #24: Change the postfix stack example to a maze example
Title

I never liked the postfix, and neither does @twentylemon, so that settles it.

### PR DESCRIPTION
**Pull request by jameshughes89**
Subsection with a postfix notation example. 

~~I don't **love** the example, but it's one of the quintessential examples.~~

postfix is now an aside and the example is the maze. This fixes #24.

### CODE REVIEW

File: source/topic5.rst [hunk 1]
```diff
@@ -7,8 +7,50 @@ Stack Operations
...
+-------------------
+
+* We as humans prefer *infix* notation for our mathematical expressions
+    * 3 + 4 * 2 = 11
+    * (7 + 2) / 3 = 3
+
+* But this is just a convention
+* We can use other notations, like *postfix* notation
+    * 3 4 2 * + = 11
+    * 7 2 + 3 / = 3
```
Comment (twentylemon) [inline source/topic5.rst:26]: Postfix is extremely uncommon. I'd clarify this with some backwards-lisp-esque expansion:
* `3 4 2 * + = (3 (4 2 *) +) = 11`
* `7 2 + 3 /  = ((7 2 +) 3 /) = 3`

May also help illustrate why the algorithm below works before they dig into the larger example.

**Review by twentylemon — COMMENTED** [review]
I never really liked the postfix operator, it just feels contrived.

I feel like backtracking (eg, maze solving) is a better illustrative algorithm for when to apply a stack. Everyone has solved a maze, it's just easier to visualize (dead end? yeet that path and try a different one). Maze solving is much harder to write code for during a lecture, but pseudocode? Why not.

File: source/topic5.rst (hunk 1 above)
Comment (jameshughes89) [reply source/topic5.rst]: Defo will add this.

**Comment by jameshughes89** [comment]
I too don't love the postfix example. 

Maze solving is Assignment 2, so I may want to avoid it. 

Either way, I will think on this example a little more. I may opt to leave it as is for now to just have it, but open an issue about thinking of a better example going forward.

**Comment by twentylemon** [comment]
I was going to write this before, but opted not to in case I was wasting my breathe. Or, fingers, as it were.

I don't really see a maze solving assignment as a deal breaker. You'd be giving very high level pseudocode, and saying "look, a stack can be used." They'd still have to figure out that the stack to use is the callstack, plus all the pathfinding nonsense.

**Comment by jameshughes89** [comment]
As of now I am planning on doing the maze example and will likely add this example as an aside.

**Comment by jameshughes89** [comment]
With the change from PostFix -> Maze, this fixes #24.

**Review by twentylemon — COMMENTED** [review]
Extremely pogged gif, brother. I may have used pull _paths_ instead of _steps_, I think it's more akin to how people solve a maze (we don't think of cells, for example). By _path_ I mean all steps taken until an intersection or a dead end, can use a squiggly line, like you would on a real maze.

On the gif, the exit cell and the popped cells will be nearly identical colours to colour blind folks. Deuteranopia is the most common, roughly 1% of the male population, dontchaknow. [Colour blind colour wheel google](https://www.google.com/search?q=colour+blind+colour+wheel&tbm=isch&ved=2ahUKEwipir-Z6YrxAhXwFFkFHd1jCMEQ2-cCegQIABAA&oq=colour+blind+colour+wheel&gs_lcp=CgNpbWcQAzIECCMQJzoECAAQQzoHCAAQsQMQQzoCCAA6BQgAELEDOggIABCxAxCDAToGCAAQChAYOgYIABAIEB46BAgAEB5Q0LcCWJfhAmCq4gJoBXAAeACAAWyIAaMWkgEEMjMuN5gBAKABAaoBC2d3cy13aXotaW1nwAEB&sclient=img&ei=N9jAYOmSCvCp5NoP3cehiAw).

**Comment by jameshughes89** [comment]
I'll leave maze for now, but will add issues for future.

**Comment by twentylemon** [comment]
> I'll leave maze for now, but will add issues for future.

#30, #31

File: source/topic5-postfix.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
+**************************
+Topic #5 Aside --- Postfix
+**************************
```
[the remark this answers is not in the document]
Comment (jameshughes89) [reply source/topic5-postfix.rst:3]: Yeah. It merged to here.

**Pull request merged**